pkg/upgrade/upgrade.go

```diff
@@ -78,29 +80,62 @@ func ManagedInstances(ctx context.Context, c client.Client, reader client.Reader
 }
 
 // ManagedInstance performs the necessary changes to bring the given Jaeger instance to the current version
-func ManagedInstance(ctx context.Context, client client.Client, jaeger v1.Jaeger) (v1.Jaeger, error) {
+func ManagedInstance(ctx context.Context, client client.Client, jaeger v1.Jaeger, latestVersion string) (v1.Jaeger, error) {
 	tracer := global.TraceProvider().GetTracer(v1.ReconciliationTracer)
 	ctx, span := tracer.Start(ctx, "ManagedInstance")
 	defer span.End()
 
-	if v, ok := versions[jaeger.Status.Version]; ok {
+	currentSemVersion, err := semver.NewVersion(jaeger.Status.Version)
+
+	if err != nil {
+		jaeger.Logger().WithFields(log.Fields{
+			"instance":  jaeger.Name,
+			"namespace": jaeger.Namespace,
+			"to":        latestVersion,
+		}).WithError(err).Warn("Skipped for update")
+	}
+	latestSemVersion, _ := semver.NewVersion(latestVersion)
+
+	if currentSemVersion.LessThan(startUpdatesVersion) {
+		// We don't know how to do an upgrade from versions lower than 1.11.0
+		jaeger.Logger().WithFields(log.Fields{
+			"instance":  jaeger.Name,
+			"namespace": jaeger.Namespace,
+			"to":        latestVersion,
+		}).Warn("Cannot automatically upgrade from versions lower than 1.11.0")
+		return jaeger, nil
+	}
+
+	if currentSemVersion.GreaterThan(latestSemVersion) {
+		// This jaeger instance has a version greater than the latest version of the operator
+		jaeger.Logger().WithFields(log.Fields{
+			"instance":  jaeger.Name,
+			"namespace": jaeger.Namespace,
+			"to":        latestVersion,
```

Comment:
Would be good to have both versions, so `"current": currentSemVersion, "latest": latestVersion,`